lib/lhm/throttler/slave_lag.rb

```diff
@@ -43,55 +45,78 @@ def throttle_seconds
         end
       end
 
-      def slave_hosts
-        slaves = get_slaves.map { |slave_host| slave_host.partition(':')[0] }
-          .delete_if { |slave| slave == 'localhost' || slave == '127.0.0.1' }
-        Lhm.logger.info "Detected slaves: #{slaves.join(',')}"
-        slaves
+      def slaves
+        @slaves[@connection] ||= get_slaves
       end
 
       def get_slaves
-        @connection.select_values(SQL_SELECT_SLAVE_HOSTS)
+        slaves = []
+        slave_hosts = master_slave_hosts
+        while slave_hosts.any? do
+          host = slave_hosts.pop
+          slave = Slave.new(host, @get_config)
+          if slaves.map(&:host).exclude?(host) && slave.connection
+            slaves << slave
+            slave_hosts.concat(slave.slave_hosts)
+          end
+        end
+        slaves
+      end
+
+      def master_slave_hosts
```

Comment:
yeah, i honestly struggled with the naming of a few of these...

it's selecting slave hosts from the master :confused: